src/main/java/org/opensearch/knn/index/codec/BasePerFieldKnnVectorsFormat.java

```diff
@@ -59,6 +86,46 @@ public KnnVectorsFormat getKnnVectorsFormatForField(final String field) {
         return formatSupplier.apply(maxConnections, beamWidth);
     }
 
+    private KnnVectorsFormat validateAndApplyQuantizedVectorsFormatForLuceneEngine(
```

Comment:
function with multiple responsibilities. Please ensure that functions have a single responsibility here otherwise the scope of function will just increase and it becomes unmanagable.

Reply:
Ack